### ISSUE TITLE
First reset promotes all envs from level 0 to level 1 despite max_init_terrain_level=0

### ISSUE THREAD

**Issue by BianCe123** [mujocolab/mjlab#1094]
## Summary

With `max_init_terrain_level=0`, all envs should start at terrain level 0. Instead, on the first reset every env is promoted to level 1, so training begins at difficulty 1 instead of the configured 0.

## Root Cause

In `ManagerBasedRlEnv._reset_idx`, `curriculum_manager.compute()` runs **before** the `mode="reset"` events that place the robot ([manager_based_rl_env.py:553-570](src/mjlab/envs/manager_based_rl_env.py#L553-570)):

```python
def _reset_idx(self, env_ids):
    self.curriculum_manager.compute(env_ids=env_ids)   # ① curriculum first
    self.sim.reset(env_ids)
    self.scene.reset(env_ids)
    self.event_manager.apply(mode="reset", ...)        # ② robot placement later
```

`terrain_levels_vel` ([curriculums.py:37-48](src/mjlab/tasks/velocity/mdp/curriculums.py#L37-48)):

```python
distance = torch.norm(
    asset.data.root_link_pos_w[env_ids, :2] - env.scene.env_origins[env_ids, :2],
    dim=1,
)
move_up = distance > terrain_generator.size[0] / 2
```

On the initial reset:
- `root_link_pos_w` is still the MJCF default (near world origin), since reset events haven't fired yet.
- `env_origins` points to distributed terrain locations (far from origin).
- `distance` exceeds `size[0]/2` (default 5.0 m), so `move_up = True` for all envs.

`update_env_origins` ([terrain_entity.py:226](src/mjlab/terrains/terrain_entity.py#L226)) then does `levels += 1 * move_up`, bumping every env from 0 to 1.

## Reproduction

- Set `max_init_terrain_level=0` on the terrain config.
- Launch velocity training and log `terrain.terrain_levels` after the first reset.
- Observed: all levels == 1. Expected: all levels == 0.

## Impact

Training starts at difficulty 1 instead of the configured 0, breaking the intended curriculum warm-up. Only affects the **first reset**; subsequent resets see the robot near its previous `env_origin` so `distance` is small and the bug is masked.

## Suggested Fix

Skip the curriculum update on the initial reset inside `terrain_levels_vel`:

```python
if env.common_step_counter == 0:
    levels = terrain.terrain_levels.float()
    return {"mean": torch.mean(levels), "max": torch.max(levels)}
```

Alternatively, base the curriculum on displacement over the previous episode (`final_pos - start_pos`) rather than instantaneous distance to `env_origin`, which would also fix the ordering dependency.

## Environment

- mjlab version: 1.5
- Python: 3.13
- Task: `velocity`
